spec/unit/configurer_spec.rb

```diff
@@ -1027,17 +1027,15 @@ def expects_neither_new_or_cached_catalog
       @agent.run
     end
 
-    it "should fallback to an empty server when failover fails" do
+    it "should error when no servers in 'server_list' are reachable" do
       Puppet.settings[:server_list] = ["myserver:123"]
       pool = Puppet::Network::HTTP::Pool.new(Puppet[:http_keepalive_timeout])
       Puppet::Network::HTTP::Pool.expects(:new).returns(pool)
       Puppet.expects(:override).with({:http_pool => pool}).yields
       Puppet.expects(:override).with({:server => "myserver", :serverport => '123'}).yields
-      Puppet.expects(:override).with({:server => nil, :serverport => nil}).yields
       error = Net::HTTPError.new(400, 'dummy server communication error')
       Puppet::Node.indirection.expects(:find).raises(error)
-      @agent.expects(:run_internal).returns(nil)
-      @agent.run
+      expect{ @agent.run }.to raise_error(Puppet::Error)
```

Comment:
Sorry just noticed this. I usually add (part of) the expected message, eg `/Could not select .... /`, so the test doesn't pass when it shouldn't.

Reply:
oh derp, yes, thank you for catching this. I meant to go back and change that

Reply:
Updated

Reply:
Cool thanks!